website/src/wasm-demo/www/index.js

```diff
@@ -0,0 +1,178 @@
+import "monaco-editor/esm/vs/editor/contrib/folding/folding";
+import "monaco-editor/esm/vs/editor/contrib/format/formatActions";
+import "monaco-editor/esm/vs/editor/contrib/goToDefinition/goToDefinitionCommands";
+import "monaco-editor/esm/vs/editor/contrib/goToDefinition/goToDefinitionMouse";
+import "monaco-editor/esm/vs/editor/contrib/gotoError/gotoError";
+import "monaco-editor/esm/vs/editor/contrib/hover/hover";
+import "monaco-editor/esm/vs/editor/contrib/inPlaceReplace/inPlaceReplace";
+import "monaco-editor/esm/vs/editor/contrib/linesOperations/linesOperations";
+import "monaco-editor/esm/vs/editor/contrib/links/links";
+import "monaco-editor/esm/vs/editor/contrib/multicursor/multicursor";
+import "monaco-editor/esm/vs/editor/contrib/parameterHints/parameterHints";
+import "monaco-editor/esm/vs/editor/contrib/referenceSearch/referenceSearch";
+import "monaco-editor/esm/vs/editor/contrib/rename/rename";
+import "monaco-editor/esm/vs/editor/contrib/smartSelect/smartSelect";
+import "monaco-editor/esm/vs/editor/contrib/snippet/snippetController2";
+import "monaco-editor/esm/vs/editor/contrib/suggest/suggestController";
+import "monaco-editor/esm/vs/editor/contrib/toggleTabFocusMode/toggleTabFocusMode";
+import "monaco-editor/esm/vs/editor/contrib/wordHighlighter/wordHighlighter";
+import "monaco-editor/esm/vs/editor/contrib/wordOperations/wordOperations";
+import "monaco-editor/esm/vs/editor/standalone/browser/accessibilityHelp/accessibilityHelp";
+import "monaco-editor/esm/vs/editor/standalone/browser/inspectTokens/inspectTokens";
+import "monaco-editor/esm/vs/editor/standalone/browser/iPadShowKeyboard/iPadShowKeyboard";
+import "monaco-editor/esm/vs/editor/standalone/browser/quickOpen/quickOutline";
+import "monaco-editor/esm/vs/editor/standalone/browser/quickOpen/gotoLine";
+import "monaco-editor/esm/vs/editor/standalone/browser/quickOpen/quickCommand";
+import "monaco-editor/esm/vs/editor/standalone/browser/toggleHighContrast/toggleHighContrast";
+
+import * as monaco from "monaco-editor/esm/vs/editor/editor.api";
+import * as rust_conf from "monaco-editor/esm/vs/basic-languages/rust/rust";
+
+import example_code from "./example-code";
+const wasm_demo = import("wasm_demo");
+
+import "./index.css";
+
+self.MonacoEnvironment = {
+    getWorkerUrl: () => "./editor.worker.bundle.js",
+};
+
+const modeId = "ra-rust" // not "rust" to circumvent conflict
+monaco.languages.register({ // language for editor
+    id: modeId,
+});
+monaco.languages.register({ // language for hover info
+    id: "rust",
+});
+
+monaco.languages.onLanguage(modeId, async () => {
+    const { WorldState } = await wasm_demo;
+
+    // let editActionId = myEditor.addCommand(0, (...args) => {
+    //     console.warn(args)
+    // })
+
+    const state = new WorldState();
+
+    const [model] = monaco.editor.getModels();
+    let allTokens = []
+
+    function update() {
+        console.warn("update")
```

Comment:
Could you please end statements with `;`? While javascript allows it, only npm doesnt use them.

Reply:
Most of the JS code is still formatted/structured badly, heck it even has warn's for logging and commented-out code.
The whole file needs some serious cleaning, but I was planning to do this after implementing most of the demo-able functionality.